src/main/java/thePackmaster/patches/gemspack/SocketGemGridPatchCancelAction.java

```diff
@@ -0,0 +1,61 @@
+package thePackmaster.patches.gemspack;
+
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePrefixPatch;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.helpers.controller.CInputActionSet;
+import com.megacrit.cardcrawl.helpers.input.InputHelper;
+import com.megacrit.cardcrawl.rooms.RestRoom;
+import com.megacrit.cardcrawl.ui.buttons.CancelButton;
+import thePackmaster.packs.GemsPack;
+import thePackmaster.vfx.gemspack.SocketGemEffect;
+
+@SpirePatch(clz = CancelButton.class, method = "update")
+public class SocketGemGridPatchCancelAction {
+
+    @SpirePrefixPatch
+    public static void Prefix(CancelButton obj) {
+        if (!obj.isHidden) {
+            obj.hb.update();
+            if (obj.hb.clicked || (InputHelper.pressedEscape || CInputActionSet.cancel.isJustPressed()) && obj.current_x != CancelButton.HIDE_X) {
```

Comment:
I understand being lazy for testing but please actually use a insert patch with a locator for `AbstractDungeon.screenSwap` here instead of this.

Reply:
This was lifted from the Guardian code verbatim, which has apparently been a lazy patch for 3 years running.  Change it shall be.